tests/test_cli_download_files.py

```diff
@@ -68,6 +68,37 @@ def test_download_files():
         os.remove(test_file)
 
 
+def test_download_files_root():
+    """Test download_files() command with root protocol."""
+    test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+    test_download_files = CliRunner()
+    test_result = test_download_files.invoke(
+        download_files, ["--recid", 3005, "--protocol", "root"]
+    )
+    assert test_result.exit_code == 0
+    assert os.path.isfile(test_file) is True
+    assert os.path.getsize(test_file) == 3644
+    assert test_result.output.endswith("\n==> Success!\n")
+    if os.path.isfile(test_file):
+        os.remove(test_file)
+
+
+def test_download_files_root_wrong():
+    """Test download_files() command with root protocol without xrootd."""
+    import mock
+    import sys
+
+    with mock.patch.dict(sys.modules, {"xrootdpyfs": None}):
+        test_download_files = CliRunner()
+        test_result = test_download_files.invoke(
+            download_files, ["--recid", 3005, "--protocol", "root"]
+        )
+        assert test_result.exit_code == 1
+        assert "xrootd is not installed on system" in test_result.output
```

Comment:
This test fails for me (and I do have xrootd locally available):

```
>           assert test_result.exit_code == 1
E           assert 0 == 1
E            +  where 0 = <Result okay>.exit_code

tests/test_cli_download_files.py:98: AssertionError
```

This is similar to the failure on Travis CI... So the mocking is not fully working?

Alternatively, you can import `xrootd_available` variable, and check based on that...

Reply:
Yes, The mocking is not working correctly, since when I saw examples over the web mocking of packages in `sys. modules` is possible but `xrootdpyfs` is not installed in `sys.modules` directly but in `python-packages`. I am looking for a workaround for it.

Reply:
@tiborsimko Now the testing of this case is working fine 🎉 . I tested with multiple ways and it seems to work out perfect.

Reply:
There is one way where it fails locally for me:

```console
$ mkvirtualenv test
$ pip install . 
$ ./run-tests.sh --check-pytest

...

    def test_download_files_root():
        """Test download_files() command with root protocol."""
        test_file = "3005/0d0714743f0204ed3c0144941e6ce248.configFile.py"
        if os.path.isfile(test_file):
            os.remove(test_file)
        test_download_files = CliRunner()
        test_result = test_download_files.invoke(
            download_files, ["--recid", 3005, "--protocol", "root"]
        )
>       assert test_result.exit_code == 0
E       assert 1 == 0
E        +  where 1 = <Result SystemExit(1)>.exit_code

tests/test_cli_download_files.py:81: AssertionError

===== 1 failed, 52 passed in 14.95s =====
```

That is, when I install with xrootd, everything works and 53 tests pass.  But when I install without xrootd, one test fails.  I would have expected the test to say: "xrootd not available, so I'm not running".  Basically you can try to import  `xrootd_available` and [skip this test](https://docs.pytest.org/en/stable/skipping.html#skipping-test-functions) when it is not applicable on the given platform.

Reply:
One more thing: Can you please enrich `docs/usage.rst` with a tiny `download-files --protocol root` example?

Reply:
@tiborsimko I think it will be more apt to use this [importskip](https://docs.pytest.org/en/stable/skipping.html#skipping-on-a-missing-import-dependency) here.